src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -398,13 +398,10 @@ function updateDOMProperties(
     } else if (propKey === DANGEROUSLY_SET_INNER_HTML) {
       var nextHtml = nextProp ? nextProp[HTML] : undefined;
       var lastHtml = lastProp ? lastProp[HTML] : undefined;
-      if (nextHtml) {
-        if (lastHtml) {
```

Comment:
Does this condition still need? It seems to be unnecessary because it is covered  by `if (lastHtml !== nextHtml)`.